ipod-shuffle-4g.py

```diff
@@ -408,7 +423,7 @@ def __init__(self, parent):
                           ("header_id", ("4s", b"hphs")), #shph
                           ("total_length", ("I", 0)),
                           ("number_of_playlists", ("I", 0)),
-                          ("number_of_non_podcast_lists", ("2s", b"\xFF\xFF")),
+                          ("number_of_non_podcast_lists", ("H", 65535)),
```

Comment:
Why did you change this?

Reply:
At the time, because I assumed I needed to specify that it was an unsigned short instead of two bytes, but I suppose they're equivalent. I do think it's useful to indicate the actual type now that it'll change values.

Reply:
To reduce the noise I would revert this change

Reply:
I just tested and actually can't revert this. Packing the structure to build the database throws an exception because the struct has a `short` but is expecting `bytes`.
~~~~
Traceback (most recent call last):
  File ".../IPod-Shuffle-4g/./ipod-shuffle-4g.py", line 810, in <module>
    shuffle.write_database()
  File ".../IPod-Shuffle-4g/./ipod-shuffle-4g.py", line 675, in write_database
    f.write(self.tunessd.construct())
  File ".../IPod-Shuffle-4g/./ipod-shuffle-4g.py", line 293, in construct
    play_header = self.play_header.construct(self.track_header.tracks)
  File ".../IPod-Shuffle-4g/./ipod-shuffle-4g.py", line 463, in construct
    output = Record.construct(self)
  File ".../IPod-Shuffle-4g/./ipod-shuffle-4g.py", line 215, in construct
    output += struct.pack("<" + fmt, self._fields.get(i, default))
struct.error: argument for 's' must be a bytes object
~~~~

Reply:
Indeed H seems correct here. Can you give it the value `0xFFFF`?

I am also wondering if a non-podcasts count makes sense. I guess a podcast count makes more sense, but that is what the docs from reverse engineering say. But maybe those are wrong!?

Reply:
Technically, `("H", b"\xFF\xFF")` would be accepted, but I think that just looks confusing, given that it's representing a number.

I also don't think I'm going to question the docs. If it was actually a "podcast count", 0 would make more sense for the case when there aren't any podcast playlists.

Reply:
0xFFFF is more clear than the number. Maybe it is also a signed short, which would be -1.